### PR TITLE
Improve General IDE support

### PR DESCRIPTION
**Pull request by larsgrefer**
see also: https://github.com/apache/tomcat/pull/282#pullrequestreview-399324287

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+root = true
+
+[*]
+end_of_line = lf
+indent_style = space
+trim_trailing_whitespace = true
+
+[*.java]
+indent_size = 4
+max_line_length = 100
```
Comment (markt-asf) [inline .editorconfig:10]: I think we have more to 120 for this.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+root = true
```
Comment (markt-asf) [inline .editorconfig:1]: There probably needs to be some discussion on tge dev@ list whether or not we want to start using this. My own first impressions are positive.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+indent_size = 4
+max_line_length = 100
+ij_java_block_brace_style = end_of_line
+ij_java_else_on_new_line = false
+
+[*.{txt,md}]
+max_line_length = 80
+
+[*.xml]
+indent_size = 2
```
Comment (markt-asf) [inline .editorconfig:18]: I'm not sure how consistent this is throughout the code base. I think the docs tend to use 2 but elsewhere it tends to be 4.

File: .editorconfig (hunk 1 above)
Comment (larsgrefer) [reply .editorconfig:10]: I took the numbers from here: https://github.com/apache/tomcat/blob/master/CONTRIBUTING.md#L127

File: .editorconfig (hunk 2 above)
Comment (larsgrefer) [reply .editorconfig:1]: More information can be found here: https://editorconfig.org/

**Comment by markt-asf** [comment]
Thanks for this. I want to do a few checks locally and then I'll back-port it.

**Pull request merged**